src/AppInstallerCLICore/Workflows/FontFlow.cpp

```diff
@@ -0,0 +1,118 @@
+#include "FontFlow.h"
+#include "TableOutput.h"
+#include <winget/Fonts.h>
+
+namespace AppInstaller::CLI::Workflow
+{
+    using namespace AppInstaller::CLI::Execution;
+
+    namespace
+    {
+        struct InstalledFontFamiliesTableLine
+        {
+            InstalledFontFamiliesTableLine(Utility::LocIndString familyName, int faceCount)
+                : FamilyName(familyName), FaceCount(faceCount) {}
+
+            Utility::LocIndString FamilyName;
+            int FaceCount;
+        };
+
+        struct InstalledFontFacesTableLine
+        {
+            InstalledFontFacesTableLine(Utility::LocIndString faceName, Utility::LocIndString familyName, std::filesystem::path filePath)
+                : FaceName(faceName), FamilyName(familyName), FilePath(filePath) {}
+
+            Utility::LocIndString FaceName;
+            Utility::LocIndString FamilyName;
+            std::filesystem::path FilePath;
+        };
+
+        void OutputInstalledFontFamiliesTable(Execution::Context& context, const std::vector<InstalledFontFamiliesTableLine>& lines)
+        {
+            Execution::TableOutput<2> table(context.Reporter, { Resource::String::FontFamily, Resource::String::FontFaceCount });
+
+            for (const auto& line : lines)
+            {
+                table.OutputLine({ line.FamilyName, std::to_string(line.FaceCount) });
+            }
+
+            table.Complete();
+        }
+
+        void OutputInstalledFontFacesTable(Execution::Context& context, const std::vector<InstalledFontFacesTableLine>& lines)
+        {
+            Execution::TableOutput<3> table(context.Reporter, { Resource::String::FontFace, Resource::String::FontFamily, Resource::String::FontFilePaths });
+
+            for (const auto& line : lines)
+            {
+                table.OutputLine({ line.FaceName, line.FamilyName, line.FilePath.u8string() });
+            }
+
+            table.Complete();
+        }
+    }
+
+    void ReportInstalledFontsResult(Execution::Context& context)
+    {
+        if (context.Args.Contains(Args::Type::FamilyName))
+        {
+            // TODO: Utilize font index for better searching capabitility.
+            const auto& familyNameArg = context.Args.GetArg(Args::Type::FamilyName);
+            const auto& fontFamily = AppInstaller::Fonts::GetInstalledFontFamily(AppInstaller::Utility::ConvertToUTF16(familyNameArg));
+
+            if (!fontFamily.has_value())
+            {
+                context.Reporter.Info() << Resource::String::NoInstalledFontFound << std::endl;
+                return;
+            }
+
+            const auto& familyName = Utility::LocIndString(Utility::ConvertToUTF8(fontFamily->Name));
+            std::vector<InstalledFontFacesTableLine> lines;
+
+            for (const auto& fontFace : fontFamily->Faces)
+            {
+                bool isFirstLine = true;
+                for (const auto& filePath : fontFace.FilePaths)
+                {
+                    if (isFirstLine)
```

Comment:
I'm not sure that I like the idea of having table lines that have some fields being empty. While it can be appealing visually, I know that someone out there will want to script against it (even when it makes its way into the PowerShell module).  

You never know what someone will do with the output either - they could pipe it to `Sort-Object` and then it would be really confusing since all the lines would be out of order and they wouldn't contain all the information necessary to stand independently

Reply:
I resorted to showing every value for each line. You bring up a good point and I'm sure someone may try to script with this table. I have yet to see a face that is supported by multiple font files so hopefully this doesn't even appear.